File: src/prod/src/Reliability/Failover/common/ReplicaStatus.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "Common.Stdafx.h"

using namespace Common;
using namespace Reliability::ReplicaStatus;

void Reliability::ReplicaStatus::WriteToTextWriter(TextWriter & w, Enum const & val)
{
    switch (val)
    {
        case Invalid: 
            w << L"I"; return;
        case Down: 
            w << L"D"; return;
        case Up: 
            w << L"U"; return;
        default: 
            Common::Assert::CodingError("Unknown Replica Status");
    }
}

::FABRIC_REPLICA_STATUS Reliability::ReplicaStatus::ConvertToPublicReplicaStatus(ReplicaStatus::Enum toConvert)
{
    switch(toConvert)
    {
    case Invalid:
        return ::FABRIC_REPLICA_STATUS_INVALID;
    case Down:
        return ::FABRIC_REPLICA_STATUS_DOWN;
    case Up:
        return ::FABRIC_REPLICA_STATUS_UP;
    default:
        Common::Assert::CodingError("Unknown Replica Status");
    }
}

File: src/prod/src/managed/Api/src/System/Fabric/ReplicaStatus.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System.Fabric.Interop;

    /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
    public enum ReplicaStatus
    {
        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        Invalid = NativeTypes.FABRIC_REPLICA_STATUS.FABRIC_REPLICA_STATUS_INVALID,

        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        Down = NativeTypes.FABRIC_REPLICA_STATUS.FABRIC_REPLICA_STATUS_DOWN,

        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        Up = NativeTypes.FABRIC_REPLICA_STATUS.FABRIC_REPLICA_STATUS_UP
    }
}